mbed_tools_ci/utils/filesystem_helpers.py

```diff
@@ -77,3 +81,56 @@ def find_file_in_tree(file_name: str,
             f'File [{file_name}] not found anywhere in directories {"above" if top else "under"} {starting_point}'
         )
     return file_path
+
+
+class TemporaryDirectory:
+    """Creates and returns a temporary directory.
+
+    This is the same as tempfile.TemporaryDirectory in most cases.
+    However, on CI the temporary folder may not be accessible.
+    Moreover, on Windows TemporaryDirectory will fail cleaning up
+    (https://bugs.python.org/issue22107) and hence, a workaround had to be
+    implemented.
+
+    """
+
+    def __init__(self, path: Optional[str] = None) -> None:

```

Comment:
I don't think this should be concerned with a path. I'd go with "Hey `TempDirectory` just give me a valid temp directory, I don't care where it is".

Reply:
I removed it but it may need to be reintroduced in the future if we have limitation in some CI systems. This would be a conversation for later